src/Apps/Artist/Routes/AuctionResults/AuctionResultsFilterContext.tsx

```diff
@@ -144,13 +166,31 @@ const AuctionResultsFilterReducer = (
       const primitiveFilterTypes: Array<keyof AuctionResultsFilters> = [
         "sort",
         "page",
+        "createdAfterYear",
+        "createdBeforeYear",
       ]
       primitiveFilterTypes.forEach(filter => {
         if (name === filter) {
           filterState[name as any] = value
         }
       })
 
+      if (name === "createdBeforeYear" && value) {
+        if (!state.createdAfterYear) {
+          filterState.createdAfterYear = `${state.earliestCreatedYear}`
+        } else if (parseInt(state.createdAfterYear) > parseInt(value)) {
+          filterState.createdAfterYear = value
+        }
+      }
+
+      if (name === "createdAfterYear" && value) {
+        if (!state.createdBeforeYear) {
+          filterState.createdBeforeYear = `${state.latestCreatedYear}`
+        } else if (parseInt(state.createdBeforeYear) < parseInt(value)) {
+          filterState.createdBeforeYear = value
+        }
+      }
```

Comment:
I added some extra filtering logic to test the cases when the filters are set in a weird way. I.e. when the earliest year is set to a time later than the latest year.